### PR TITLE
feat(misconf): Add support for `--include-deprecated-checks`

### PR DESCRIPTION
**Pull request by simar7**
## Description

Adds support for `--include-deprecated-checks`. This is a stacked PR on https://github.com/aquasecurity/trivy/pull/6664

## Related issues
- Closes https://github.com/aquasecurity/trivy/issues/6665

## Related PRs
- [ ] https://github.com/aquasecurity/trivy/pull/6664

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,7 @@ require (
 	github.com/aquasecurity/table v1.8.0
 	github.com/aquasecurity/testdocker v0.0.0-20240419073403-90bd43849334
 	github.com/aquasecurity/tml v0.6.1
-	github.com/aquasecurity/trivy-aws v0.8.0
+	github.com/aquasecurity/trivy-aws v0.8.1-0.20240511051125-4393910b056b
```
Comment (simar7) [inline go.mod:26]: note: update this when https://github.com/aquasecurity/trivy-aws/pull/148 is merged

**Comment by simar7** [comment]
@nikpivkin we can merge this first once you review it as it stacks on top of https://github.com/aquasecurity/trivy/pull/6664

File: pkg/iac/rego/scanner.go [hunk 2]
```diff
@@ -248,6 +253,12 @@ func (s *Scanner) ScanInput(ctx context.Context, inputs ...Input) (scan.Results,
 			continue
 		}
 
+		if !s.includeDeprecatedChecks {
```
Comment (nikpivkin) [inline pkg/iac/rego/scanner.go:256]: conditions can be combined

**Review by nikpivkin — APPROVED** [review]
left a small comment, but lgtm

**Pull request merged**